### PR TITLE
xbps-create: do not add a build-date property to packages.

### LINKED ISSUE #110: reproducable builds support
Currently it is not possible with xbps to produce the same packages from the same source.

1.  `build-date` is different for every run of `xbps-create(1)` as it is set for the build time of the package. I think it's a bad idea to set the build-date staticly as that would be missleading.

  What side effects could be occur if we simply remove build-date? The build date is still coded in the mtime/ctime of the `.xbps` package.

2. Maybe there are some other issues such as the libarchive version that would produce different checksums in the resulting package.

### PR DESCRIPTION
**Pull request by Gottox**
Removes the build-date property as discussed in #110. closes #110.

### CODE REVIEW

**Comment by xtraeme** [comment]
I agree with this change but IMHO it's incomplete. Why?
  - removing this object means that we cannot know anymore what pkgs were built at what date

I'd like to see a change to `xbps-rindex(1)` to `fstat()` the generated binpkg and set the `build-date` object to the repository index instead.

This way we won't lose any metadata and builds will still be reproducable.

Thanks

**Comment by Gottox** [comment]
okay.

File: bin/xbps-rindex/index-add.c [hunk 1]
```diff
@@ -188,6 +206,12 @@ index_add(struct xbps_handle *xhp, int args, int argmax, char **argv, bool force
 			rv = EINVAL;
 			goto out;
 		}
+		if (set_build_date(binpkgd, st.st_mtime) > 0) {
```
Comment (xtraeme) [inline bin/xbps-rindex/index-add.c]: `set_build_date()` returns 0 or -1, please fix.

**Comment by Gottox** [comment]
fixed

**Comment by xtraeme** [comment]
@Gottox thanks!

**Comment by xtraeme** [comment]
@Gottox perfect! please merge, thanks :+1:

**Pull request merged**